plasmapy/formulary/shocks.py

```diff
@@ -0,0 +1,106 @@
+""" This module gathers functions relating to shocks and the properties thereof.
+"""
+__all__ = ["entropy_across_shock_polytropic", "ds_"]
+
+
+import warnings
+from astropy import units as u
+from plasmapy.utils.exceptions import PhysicsWarning
+from plasmapy.utils.decorators import (
+    angular_freq_to_hz,
+    check_relativistic,
+    validate_quantities,
+)
+from astropy.constants import c, a0, k_B
+from numpy import pi, exp, sqrt, log, nan, isclose
+
+
+@validate_quantities(
+    rho_1={"can_be_negative": False},
+    rho_2={"can_be_negative": False},
```

Comment:
Since you're immediately making ratios from `p_1`, `p_2`, `rho_1`, and  `rho_2`, this is a good instance to use `pass_equivalent_units=True`.  This will keep your function from doing 2 extra conversions.  This is not a big deal for `p`, but can be for `rho` because the the conversion to `u.m ** -3` can cause non-negligible floating point errors.

Reply:
do I just use `pass_equivalent_units=True` in `@validate_quantities`?

Reply:
You add the key-value pair to the dictionary.  There's an example in the [`validate_quantities` documentation](https://docs.plasmapy.org/en/latest/api/plasmapy.utils.decorators.validate_quantities.html#plasmapy.utils.decorators.validate_quantities).